src/lib/common/sol-mainloop-impl-posix.c

```diff
@@ -786,6 +786,28 @@ sol_mainloop_impl_fd_del(void *handle)
     return true;
 }
 
+bool
+sol_mainloop_impl_fd_set_flags(void *handle, unsigned int flags)
+{
+    struct sol_fd_posix *fd_handle = handle;
+
+    SOL_NULL_CHECK(handle, false);
+
+    fd_handle->flags = flags;
+
```

Comment:
@ulissesf remembered we may need to lock the main loop and notify the main thread so `poll()` will run again if modified the fd from a different thread.